storage/tool/src/main/java/com/hortonworks/registries/storage/tool/sql/Utils.java

```diff
@@ -32,16 +33,27 @@
         return objectMapper.readValue(new File(configFilePath), Map.class);
     }
 
-    public static void loadJarIntoClasspath(File jarFile) throws Exception {
+    public static ClassLoader loadJarIntoClasspath(final File jarFile) throws Exception {
         try {
-            URL url = jarFile.toURI().toURL();
-            URLClassLoader classLoader = (URLClassLoader) ClassLoader.getSystemClassLoader();
-            Method method = URLClassLoader.class.getDeclaredMethod("addURL", URL.class);
-            method.setAccessible(true);
-            method.invoke(classLoader, url);
-        } catch(Exception e) {
+            final MyURLClassLoader customClassLoader = AccessController.doPrivileged((PrivilegedAction<MyURLClassLoader>) () ->
+                    new MyURLClassLoader(new URL[0], Thread.currentThread().getContextClassLoader()));
+            final URL url = jarFile.toURI().toURL();
+            customClassLoader.addUrl(url);
+            return customClassLoader;
+        } catch (final Exception e) {
             System.out.println("Failed to load " + jarFile + " into classpath.");
             throw new Exception(e);
         }
     }
+
+    static class MyURLClassLoader extends URLClassLoader {
```

Comment:
May be we can call this JarClassLoader or some other better name ?

Reply:
Renamed to JarClassLoader.